### DISCUSSION TITLE
Subscription to selection event

### DISCUSSION THREAD

**Discussion by Marwa-Da** [molstar/molstar#1008] [Q&A] [answered]
Hi all,

I want to be notified when a selection is appliedregardless of whether it's done programmatically or through the Molstar UI.
 I used this ` PluginCommands.Interactivity.Structure.Select.subscribe(this.plugin, handler)` However, the handler is not being called.

Could you tell me what am I missing?

**Comment by dsehnal** [comment, marked as answer]
`plugin.behaviors.interaction.*.subscribe(...)`

What you are calling is a command to trigger the event.

**Comment by Marwa-Da** [comment]
So that also applies to all PluginCommands such as `PluginCommands.Layout.Update.subscribe` too, right?
I'm asking again because I subscribed to it and the handler is being called.
I think I'm still not really getting the plugin commands idea! :((

**Reply by dsehnal** [reply]
`PluginCommand`s are generally used by the UI to trigger things. You can also trigger them yourself by just calling the command on the `PluginContext`; or subscribe to them to do custom things. This pattern allows overriding (certain) UI actions on per plugin instance basis.
